### PR TITLE
tests: Print dots by default in functional tests

### PR DESCRIPTION
**Pull request by ken2812221**
In cron job (https://travis-ci.org/bitcoin/bitcoin/builds/445823485), the functional tests would fail due to silent for 10 mins.

After applying this patch, we con't see any extra characters printed on screen but also avoid timeout (https://travis-ci.org/ken2812221/bitcoin/builds/445981698)

### CODE REVIEW

**Comment by laanwj** [comment]
For this reason the test runner prints `.` every so few seconds, but somehow—that's no longer enough?

**Comment by MarcoFalke** [comment]
This is probably due to `if self.logging_level == logging.DEBUG:` from #14504. Imo that condition can be removed.

**Comment by ken2812221** [comment]
@MarcoFalke Is this what you mean or just remove the `if` condition to always print `.`?

**Comment by MarcoFalke** [comment]
Previously there was no condition, and it would always print: https://github.com/bitcoin/bitcoin/pull/14504/files?w=1#diff-a5b9b84e3a3387476629e74ddb227a7eR502

Not sure why that got added.

**Comment by MarcoFalke** [comment]
@isghe

**Comment by Gnappuraz** [comment]
@ken2812221 don't you have to kill the waiting at some point since you put it on background?
Also I think that travis has a 10m timeout, so printing every 9m should cut it.
I've done something like this to avoid it while running the tests... 

```if [ "$RUN_TESTS" = "true" ]; then
  while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
  make $MAKEJOBS check VERBOSE=1
  kill %1
fi```

**Comment by isghe** [comment]
> Previously there was no condition, and it would always print: https://github.com/bitcoin/bitcoin/pull/14504/files?w=1#diff-a5b9b84e3a3387476629e74ddb227a7eR502
> 
> Not sure why that got added.

To be coherent with `--quiet` option "only print results summary and failure logs".

@ken2812221 Maybe we can add the `--dots` option to show the dots. So the travis command could be
`DOCKER_EXEC test/functional/test_runner.py --combinedlogslen=4000 --coverage --quiet --failfast --dots ${extended}`.

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -495,14 +494,7 @@ def get_next(self):
...
                     self.num_running -= 1
                     self.jobs.remove(job)
-                    if self.logging_level == logging.DEBUG:
-                        clearline = '\r' + (' ' * dot_count) + '\r'
-                        print(clearline, end='', flush=True)
-                        dot_count = 0
                     return TestResult(name, status, int(time.time() - start_time)), testdir, stdout, stderr
-            if self.logging_level == logging.DEBUG:
-                print('.', end='', flush=True)
-                dot_count += 1
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]: I think it was nice to have some indicator that something was running. I'd prefer to keep it.

Comment (isghe) [reply test/functional/test_runner.py]: I agree: running functional tests, without any progress indicator, it's not a good incentive to run functional tests.

**Review by Gnappuraz — APPROVED** [review]
utACK 2e4435c

File: test/functional/test_runner.py [hunk 2]
```diff
@@ -441,7 +443,7 @@ class TestHandler:
     Trigger the test scripts passed in via the list.
     """
 
-    def __init__(self, num_tests_parallel, tests_dir, tmpdir, test_list=None, flags=None, logging_level=logging.DEBUG):
+    def __init__(self, num_tests_parallel, tests_dir, tmpdir, test_list=None, flags=None, logging_level=logging.DEBUG, dots=False):
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]: `self.logging_level` is now unused. I'd slightly prefer to remove it.

File: test/functional/test_runner.py [hunk 3]
```diff
@@ -225,6 +225,7 @@ def main():
     parser.add_argument('--quiet', '-q', action='store_true', help='only print results summary and failure logs')
     parser.add_argument('--tmpdirprefix', '-t', default=tempfile.gettempdir(), help="Root directory for datadirs")
     parser.add_argument('--failfast', action='store_true', help='stop execution after the first test failure')
+    parser.add_argument('--dots', action='store_true', help='print dots while testing to avoid silence timeout on travis-ci.')
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]: Is it really necessary to add an option to enable a feature that should be on by default? I think the average user would prefer to see dots without having to enable them by typing `--dots` every time.

File: test/functional/test_runner.py [hunk 4]
```diff
@@ -225,6 +225,7 @@ def main():
     parser.add_argument('--quiet', '-q', action='store_true', help='only print results summary and failure logs')
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]:
```suggestion
    parser.add_argument('--quiet', '-q', action='store_true', help='only print dots, results summary and failure logs')
```

Comment (MarcoFalke) [reply test/functional/test_runner.py]: Could just add dots here and remove the other option?

**Review by maflcko — COMMENTED** [review]
Sorry for nitpicking this trivial change to death. Feel free to ignore my feedback.

**Comment by ken2812221** [comment]
@MarcoFalke Now it print dots by default.

**Comment by MarcoFalke** [comment]
utACK 4bd125fff080d1d1f20d0259836d55bb685be6d2

**Comment by fanquake** [comment]
utACK 4bd125f

**Comment by jnewbery** [comment]
Seems reasonable. The whole point of quiet mode was to not fill the screen with output. Now that the dots erase themselves after each test, that's no longer an issues.

Tested ACK 4bd125fff080d1d1f20d0259836d55bb685be6d2

**Pull request merged**